### PR TITLE
tests: replace while commands with the retry tool

### PR DESCRIPTION
**Pull request by sergiocazzolato**
The idea is to avoid these infinite loops and use the retry tool instead.

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Thanks

**Review by anonymouse64 — APPROVED** [review]
lgtm, thanks for this

File: tests/core/failover/task.yaml [hunk 1]
```diff
@@ -116,13 +116,7 @@ execute: |
...
     # after rollback, we have the original target snap for a while
     # wait until the kernel and core snap revisions are in place
-    while true ; do
-        current=$(snap list | awk "/^${TARGET_SNAP_NAME} / {print(\$3)}")
-        if [ "$current" = "$(cat prevBoot)" ] ; then
-            break
-        fi
-        sleep 1
-    done
+    retry -n 60 --wait 1 sh -c "test $(snap list | awk "/^${TARGET_SNAP_NAME} / {print(\$3)}") = $(cat prevBoot)"
```
Comment (bboozzoo) [inline tests/core/failover/task.yaml:119]: Hm I think this is wrong, $(snap list | awk ..) will only be evaluated once when expanding the command. The correct version likely looks like this:
```
#shellcheck disable=<somecheck>
TARGET_SNAP_NAME="$TARGET_SNAPNAME"  \
    retry -n 60 --wait 1 sh -c 'test $(snap list | awk "/^${TARGET_SNAP_NAME} / {print(\$3)}") = $(cat prevBoot)'
```

**Review by bboozzoo — CHANGES_REQUESTED** [review]

File: tests/core/failover/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/core/failover/task.yaml:119]: @bboozzoo you are right, thanks for the review. Fix in progress

**Review by bboozzoo — APPROVED** [review]
LGTM. Let's see if the tests are happy too.

File: tests/core/failover/task.yaml (hunk 1 above)
Comment (bboozzoo) [reply tests/core/failover/task.yaml]:
```
 TARGET_SNAP_NAME="$TARGET_SNAP_NAME" retry -n 60 --wait 1 sh -c 'test $(snap list | awk "/^${TARGET_SNAP_NAME} / {print(\$3)}") = $(cat prevBoot)'
                                                                ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.
```
we need to explicitly disable SC2016

**Pull request merged**